### PR TITLE
Brings DNA Console behaviour server-side to prevent href exploitation.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Fixes the issue #65498 addresses in a more complete way, by moving the logic for selecting the genes from the client side to the server side.

Previously the tgui interface would tell DM what the next gene was. Now the interface tells DM what action the user performed and DM selects the appropriate gene based on the action.

Functionality is still the same.

## Why It's Good For The Game

Href exploits bad. Hotfixes good. Complete fixes better.

### CODE REVIEW

File: code/game/machinery/computer/dna_console.dm [hunk 1]
```diff
@@ -40,6 +40,16 @@
...
 
+/// Input from tgui interface. X the gene out.
+#define CLEAR_GENE 0
+/// Input from tgui interface. Progress to the next gene.
+#define NEXT_GENE 1
+/// Input from tgui interface. Progress to previous gene.
+#define PREV_GENE 2
+
+/// List of genes in sequence for iterating.
+#define GENE_LETTERS list("A", "T", "C", "G");
```
Comment (Mothblocks) [inline code/game/machinery/computer/dna_console.dm:51]: I don't think this should be a define at all, you only use it in one place, and you name it `gene_letters` again directly before

File: code/game/machinery/computer/dna_console.dm [hunk 2]
```diff
@@ -505,24 +518,35 @@
...
-				newgene = truegenes[genepos]
-				jokerready = world.time + JOKER_TIMEOUT - (JOKER_UPGRADE * (connected_scanner.precision_coeff-1))
+			if(genepos > length(sequence))
+				CRASH("Unexpected input for \[\"pos\"\] param sent to [type] tgui interface. Consult tgui logs for error.")
 
-			// If the gene is an X, we want to update the default genes with the new
-			//  X to allow highlighting logic to work on the tgui interface.
-			if(newgene == "X")
+			// X out the gene.
+			if(pulse_action == CLEAR_GENE)
```
Comment (Mothblocks) [inline code/game/machinery/computer/dna_console.dm:529]: `switch`

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Some other quips in my head but I don't care enough to track them

File: code/game/machinery/computer/dna_console.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/game/machinery/computer/dna_console.dm:51]: ~~list defines are a bad pattern anyway~~

File: code/game/machinery/computer/dna_console.dm (hunk 2 above)
Comment (Timberpoes) [reply code/game/machinery/computer/dna_console.dm]: Switchified

File: code/game/machinery/computer/dna_console.dm (hunk 1 above)
Comment (Timberpoes) [reply code/game/machinery/computer/dna_console.dm]: Moved to the static list definition.

**Pull request merged**